mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveRunnerWrite.java

```diff
@@ -0,0 +1,142 @@
+    helper = new TestHelper(conf,
+        new HadoopTables(conf),
+        temp.newFolder(FileFormat.PARQUET.name()).toString(),
+        SCHEMA,
+        null,
+        FileFormat.PARQUET,
+        temp);
+
+    table = helper.createUnpartitionedTable();
+
+    shell.executeQuery("CREATE TABLE withShell STORED BY '" + HiveIcebergStorageHandler.class.getName() + "' " +
+        "LOCATION '" + table.location() + "' " +
+        "TBLPROPERTIES ('" + InputFormatConfig.WRITE_FILE_FORMAT + "'='" + FileFormat.PARQUET.name() + "')");
+  }
+
+  @After
+  public void after() {
+    shell.executeQuery("DROP TABLE withShell");
+  }
+
+  @Test
+  public void testInsert() {
+    List<Record> records = helper.generateRandomRecords(2, 0L);
+    // The expected query is like
+    // INSERT INTO withShell VALUES ('farkas', 1), ('kutya', 2)
+    StringBuilder query = new StringBuilder().append("INSERT INTO withShell VALUES ");
+    records.forEach(record -> query.append("('").append(record.get(0)).append("',").append(record.get(1)).append("),"));
+    query.setLength(query.length() - 1);
+
+    shell.executeQuery(query.toString());
+
+    HiveIcebergSerDeTestUtils.validate(table, records, 1);
+  }
+
+  @Test
+  public void testInsertFromSelect() throws IOException {
+    List<Record> records = helper.generateRandomRecords(2, 0L);
+    helper.appendToTable(null, records);
+
+    // Just check the initial data
+    HiveIcebergSerDeTestUtils.validate(table, records, 1);
+
+    shell.executeQuery("INSERT INTO withShell SELECT * FROM withShell");
+
+    // Check that everything is duplicated as expected
+    records.addAll(records);
+    HiveIcebergSerDeTestUtils.validate(table, records, 1);
+  }
+
+  @Test
+  public void testInsertFromSelectWithOrderBy() throws IOException {
+    // We expect that there will be Mappers and Reducers here
+    List<Record> records = helper.generateRandomRecords(2, 0L);
+    helper.appendToTable(null, records);
+
+    // Just check the initial data
+    HiveIcebergSerDeTestUtils.validate(table, records, 1);
+
+    shell.executeQuery("INSERT INTO withShell SELECT * FROM withShell ORDER BY id");
+
+    // Check that everything is duplicated as expected
+    records.addAll(records);
+    HiveIcebergSerDeTestUtils.validate(table, records, 1);
+  }
+
+  @Test
+  public void testDefaultFileFormat() {
+    shell.executeQuery("CREATE TABLE withShell2 STORED BY '" + HiveIcebergStorageHandler.class.getName() + "' " +
+        "LOCATION '" + table.location() + "'");
+
+    List<Record> records = helper.generateRandomRecords(2, 0L);
+    // The expected query is like
+    // INSERT INTO withShell VALUES ('farkas', 1), ('kutya', 2)
+    StringBuilder query = new StringBuilder().append("INSERT INTO withShell2 VALUES ");
+    records.forEach(record -> query.append("('").append(record.get(0)).append("',").append(record.get(1)).append("),"));
+    query.setLength(query.length() - 1);
+
+    shell.executeQuery(query.toString());
```

Comment:
Don't you want to query this back to check they were inserted correctly?

Reply:
Done - my first thought was that it is already covered by another test, but you are right - the test spinup is the time, and not the checking. So added :)